src/util/c_types.h

```diff
@@ -274,6 +274,17 @@ class c_enum_typet : public type_with_subtypet
 
   typedef std::vector<c_enum_membert> memberst;
 
+  c_enum_typet(typet _subtype, memberst enum_members)
+    : c_enum_typet(std::move(_subtype))
+  {
+    members() = std::move(enum_members);
+  }
+
+  memberst &members()
+  {
+    return (memberst &)(add(ID_body).get_sub());
+  }
+
   const memberst &members() const
   {
     return (const memberst &)(find(ID_body).get_sub());
```

Comment:
I know you didn't write those, but wouldn't a C++ style cast (either a `static_cast` or `reinterpret_cast` be a better idea in both of the `members()` methods?

Reply:
This would nominally be a `static_cast`, for the usual casting pattern we use with `irept`s.

Reply:
I was mistaken. The correct cast to use here is `reinterpret_cast`. `static_cast` work work for casting around the `irept` heirarchy directly, however the types here are two different specialisations of `std::vector`, rather than two types in the same hierarchy. Hence the `reinterpret_cast` is needed.